File: backend/AssetIQAI.Domain/Entities/Product.cs
using AssetIQAI.Domain.Common;

namespace AssetIQAI.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;

    public string SKU { get; set; } = string.Empty;

    public string? Description { get; set; }

    public Guid CategoryId { get; set; }

    public Category Category { get; set; } = null!;

    public Guid SupplierId { get; set; }

    public Supplier Supplier { get; set; } = null!;

    public decimal UnitPrice { get; set; }

    public int StockQuantity { get; set; }

    public int MinimumStock { get; set; }

    public string? ImageUrl { get; set; }

    public bool IsActive { get; set; } = true;
}

File: frontend/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  sku: string;
  description?: string;

  categoryId: string;
  categoryName?: string;

  supplierId: string;
  supplierName?: string;

  unitPrice: number;

  stockQuantity: number;

  minimumStock: number;

  imageUrl?: string;

  isActive: boolean;

  createdAt: string;

  updatedAt: string;
}

export interface ProductListResponse {
  items: Product[];

  totalCount: number;

  page: number;

  pageSize: number;

  totalPages: number;
}

export interface CreateProductDto {
  name: string;

  sku: string;

  description?: string;

  categoryId: string;

  supplierId: string;

  unitPrice: number;

  stockQuantity: number;

  minimumStock: number;

  imageUrl?: string;

  isActive: boolean;
}

export interface UpdateProductDto{
name: string;

  sku: string;

  description?: string;

  categoryId: string;

  supplierId: string;

  unitPrice: number;

  stockQuantity: number;

  minimumStock: number;

  imageUrl?: string;

  isActive: boolean;
}